lib/rules/no_but_in_given_when.js

```diff
@@ -0,0 +1,66 @@
+const { isEmpty: _isEmpty, has: _has, keys: _keys } = require("lodash");
+const { format } = require("util");
+const Rule = require("./Rule");
+
+module.exports = class NoButInGivenWhen extends Rule {
+    static meta = {
+        ruleId: "no_but_in_given_when",
+        message: "Do not use But in %s step",
+        type: "warn",
+        location: {},
+        hasFix: false,
+    };
+
+    constructor(ast, config) {
+        super(ast, config);
+    }
+
+    static run(ast, config) {
+        if (_isEmpty(ast?.feature)) return [];
+
+        return new NoButInGivenWhen(ast, config).execute();
+    }
+
+    execute() {
+        this.checkWhenStep(this._ast);
+
+        return this.getProblems();
+    }
+
+    checkWhenStep(astObject) {
```

Comment:
```suggestion
    checkGivenOrWhenStep(astObject) {
```

Also rename this, otherwise it might be confusing in future.